src/layers/legacy/medImageIO/medInrimageImageIO.cpp

```diff
@@ -0,0 +1,1172 @@
+
+
+    if (isGz)
+        ::gzclose(m_file);
+    else
+        fclose((FILE*)m_file);
+
+}
+
+
+
+/************************************************************
+     *
+     * private methods
+     *
+     ************************************************************/
+
+void
+InrimageImageIO::SwapBytesIfNecessary(void* buffer,
+                                      unsigned long numberOfPixels)
+{
+    if (ImageIOBase::GetByteOrder() == LittleEndian)
+    {
+        switch (m_PixelType)
+        {
+            case CHAR:
+                itk::ByteSwapper<char>::SwapRangeFromSystemToLittleEndian((char*)buffer,
+                                                                     numberOfPixels);
+                break;
+            case UCHAR:
+                itk::ByteSwapper<unsigned char>::SwapRangeFromSystemToLittleEndian
+                        ((unsigned char*)buffer, numberOfPixels);
+                break;
+            case SHORT:
+                itk::ByteSwapper<short>::SwapRangeFromSystemToLittleEndian
+                        ((short*)buffer, numberOfPixels);
+                break;
+            case USHORT:
+                itk::ByteSwapper<unsigned short>::SwapRangeFromSystemToLittleEndian
+                        ((unsigned short*)buffer, numberOfPixels);
+                break;
+            case INT:
+                itk::ByteSwapper<int>::SwapRangeFromSystemToLittleEndian
+                        ((int*)buffer, numberOfPixels);
+                break;
+            case UINT:
+                itk::ByteSwapper<unsigned int>::SwapRangeFromSystemToLittleEndian
+                        ((unsigned int*)buffer, numberOfPixels);
+                break;
+            case LONG:
+                itk::ByteSwapper<long>::SwapRangeFromSystemToLittleEndian
+                        ((long*)buffer, numberOfPixels);
+                break;
+            case ULONG:
+                itk::ByteSwapper<unsigned long>::SwapRangeFromSystemToLittleEndian
+                        ((unsigned long*)buffer, numberOfPixels);
+                break;
+            case FLOAT:
+                itk::ByteSwapper<float>::SwapRangeFromSystemToLittleEndian((float*)buffer,
+                                                                      numberOfPixels);
+                break;
+            case DOUBLE:
+                itk::ByteSwapper<double>::SwapRangeFromSystemToLittleEndian
+                        ((double*)buffer, numberOfPixels);
+                break;
+            default:
+                itk::ExceptionObject exception(__FILE__, __LINE__);
+                exception.SetDescription("Pixel Type Unknown");
+                throw exception;
+        }
+    }
+    else
+    {
+        switch (m_PixelType)
+        {
+            case CHAR:
+                itk::ByteSwapper<char>::SwapRangeFromSystemToBigEndian((char *)buffer,
+                                                                  numberOfPixels);
+                break;
+            case UCHAR:
```

Comment:
I get many warnings like this: /Users/ocommowi/medinria/medInria-public/src/layers/legacy/medImageIO/medInrimageImageIO.cpp:1130:18: warning: comparison of two values with different enumeration types in switch statement ('itk::ImageIOBase::IOPixelType' and 'itk::ImageIOBase::IOComponentType')

30 of them in the file, that's a lot. And I think it is actually ugly. To investigate